src/uct/ib/base/ib_md.c

```diff
@@ -1392,6 +1395,23 @@ static ucs_status_t uct_ib_query_md_resources(uct_component_t *component,
         num_resources++;
     }
 
+    if (num_resources != 0) {
+        status = ucs_sys_get_effective_memlock_rlimit(&memlock_limit);
+        if ((status == UCS_OK) &&
+            (memlock_limit <= UCT_IB_MD_REQUIRED_MEMLOCK_LIMIIT)) {
+            /* Disable the RDMA devices because of too strict locked memory limit*/
+            ucs_diag("RDMA transports are disabled because max locked memory "
+                     "limit (%llu kbytes) is too low. Please set max locked "
+                     "memory (ulimit -l) to 'unlimited'",
+                     (memlock_limit / UCS_KBYTE));
+
+            *resources_p     = NULL;
+            *num_resources_p = 0;
+            ucs_free(resources);
```

Comment:
we changed that behavior because of several reports by different users, that UCX does not use IB transport even though it is available on the system. Maybe initialize the device, but print warning/error instead?
`ucs_diag` is disabled by default, users will not notice the reason for IB mailfunctioning

Reply:
other users complain that we print this error when they don't want to use RDMA.
IMO ucs_diag is a good enough solution for a case things still work but maybe with worse performance

Reply:
i was analysing two reports where users complained about missing ib transports (without any warning sign)

Reply:
we should probably make habit of using UCX_LOG_LEVEL=diag for any issue as 1st debug measure

Reply:
ok, this makes sense. I though about making threshold configurable (so we could try 0 while analysing), but we also can just notice the error with diag trace

Reply:
I actually agree with @brminich - imho this patch will cause more harm than good. In this particular bug report, they used HFI NIC. I recently pushed changes to master that check that the device is IB or ROCE. If they have NIC that reports IB or ROCE but for whatever reason `ibv_create_cq` is not supported, this is a driver problem. Please ask them to check with the master and if they still have problem, I believe it is an issue in driver reporting invalid transport.

Reply:
https://github.com/openucx/ucx/commit/f848eea3fff3146ef1539b45d72a595136858399

Reply:
@shamisp this patch didn't helped at least one of customers. Also, I have reproduced it on the system with IB device even if it wasn't used. So do you still think that we shouldn't merge this patch?

Reply:
Is it real scenario or artificial ? In a single setup mode I get better BW when both IB and SM are used for process to process communication on the same node. I have recently pushed a patch to MD code that disables verbs device that is not ROCE or IB transport.  I think, this potentially can be expanded to check vendor_id and if the specific device does not require memory pinning, the warning can be disabled.

Reply:
I just ran one of osu benchmark on the system with IB device and memlock limit set to 2 KB. The error was printed, but benchmark executed successfully. As I understand, one of customers in the issue complains about the similar scenario. 

Ok, will try to apply this logic to the patch.

Reply:
@yosefe @shamisp It seems like little misunderstanding happens. In the issue we have two errors reported by two different people:

1. `ERROR ibv_create_cq(cqe=4096) failed: Operation not supported`, Seems to be unrelated to locked memory limit.
2. `ERROR mlx5_0: ibv_create_cq() failed: Cannot allocate memory. Please set max locked memory (ulimit -l) to 'unlimited' (current: 64 kbytes)`. Related to memory lock limit, but in that case it seems like customer has RoCE device with disabled port, so it is OK that https://github.com/openucx/ucx/commit/f848eea3fff3146ef1539b45d72a595136858399 didn't help. And also it seems OK that we report error after `ibv_create_cq` call because this device require memory pinning and reports about it.

I see the following ways of this problem solution:
1. Apply this patch to prevent errors during `ibv_create_cq` call.
1. Leave the logic as is, but provide programable way to disable current.
1. Try to understand whether device has active ports or not and do not use it if all ports are disabled.

What do you prefer?

Reply:
I agree that there is some confusion is going on.

If all ports are in down mode, we shell probably completely ignore the nic and not to try to create CQ.